### PR TITLE
update trove classifiers and tests for Python 3.9 + fix pytest config

### PR DESCRIPTION
**Pull request by terencehonles**
## Description

This change adds the trove classifiers for Python 3.8 & 3.9, updates the CI to test Python 3.9, and updates the pytest config to not try running the test setup as a test itself.

## For reviewers

- Check that the PR title is short, concise, and will make sense 1 year
  later.
- Check that new functions are imported in corresponding `__init__.py`.
- Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.

### CODE REVIEW

**Comment by terencehonles** [comment]
It appears the azure pipelines were not configured correctly and looking at the logs the Python version specified in the UsePythonVersion task did not match the Python used in the later stages. I adjusted the pipelines to explicitly refer to the `$(stage.pythonLocation)` variable (using a bash variable for brevity) and also updated bash to be run with `set -e` to abort on errors.

It looks like there are still some issues with the Python 3.9 runners and I'll see if I can figure out what the issue is tomorrow.

**Comment by emmanuelle** [comment]
Thanks a lot @terencehonles ; there seem to be indeed a few problems with some of our dependencies (Pywavelets, SimpleITK).

**Comment by hmaarrfk** [comment]
Could you revert the portion 3/.9 changes? This pr looks good otherwise and it l would be a Shane to hold it up.

**Comment by grlee77** [comment]
> Thanks a lot @terencehonles ; there seem to be indeed a few problems with some of our dependencies (Pywavelets, SimpleITK).

We do not currently have Python 3.9 wheels for PyWavelets, but I don't think it should be too much work to get those out of it is holding things up here. I can take a look sometime today/tomorrow

**Comment by terencehonles** [comment]
> > Thanks a lot @terencehonles ; there seem to be indeed a few problems with some of our dependencies (Pywavelets, SimpleITK).
> 
> We do not currently have Python 3.9 wheels for PyWavelets, but I don't think it should be too much work to get those out of it is holding things up here. I can take a look sometime today/tomorrow

Thanks @grlee77 the wheels will build on Linux, but it looks like there's something missing on Windows. If you get to it great, otherwise I was going to look at making sure it has a pyproject.toml as I noticed there were warnings for that from pip. Looking at the error on Azure Pipelines I don't think adding the pyproject.toml will actually resolve the issue though so either building a wheel so it doesn't need to be built here or identifying what needs to be fixed on Windows would be great.

**Comment by terencehonles** [comment]
> Could you revert the portion 3/.9 changes? This pr looks good otherwise and it l would be a Shane to hold it up.

@hmaarrfk I put the non Python 3.9 changes in #5054

**Comment by hmaarrfk** [comment]
Thanks @terencehonles I think you were hitting problems with compiling `imread`. That is kinda standard and why we placed them as "extra", we should be able to disable them with a flag.

**Comment by terencehonles** [comment]
> Thanks @terencehonles I think you were hitting problems with compiling `imread`. That is kinda standard and why we placed them as "extra", we should be able to disable them with a flag.

Would you prefer I disable the optional section rather than building them from source? I believe I identified the couple lines to enable that.

**Comment by terencehonles** [comment]
Nevermind on that question, it's pretty slow to build and it doesn't really make sense to make the build take that long.

**Comment by grlee77** [comment]
It looks like we also still have Appveyor builds (`.appveyor.yml`)? I just added 3.9 support on Appveyor for PyWavelets and I think you have to specify a "Visual Studio 2019" image as done here:

https://github.com/PyWavelets/pywt/blob/e2b05eaf849d516be02a187fd014d42f8df940d5/appveyor.yml#L18-L19

I will look into the Azure failure now and see if it is a similar issue there

**Comment by grlee77** [comment]
Azure passed just by restarting the job. It looks the change here now need to be rebased, though

**Comment by terencehonles** [comment]
> It looks like we also still have Appveyor builds (`.appveyor.yml`)? I just added 3.9 support on Appveyor for PyWavelets and I think you have to specify a "Visual Studio 2019" image as done here:
> 
> https://github.com/PyWavelets/pywt/blob/e2b05eaf849d516be02a187fd014d42f8df940d5/appveyor.yml#L18-L19
> 
> I will look into the Azure failure now and see if it is a similar issue there

It looks like I hadn't pushed my change which update appveyor.yml. I've updated it, but was going to follow up on all/any build errors. Thanks for the reference to the PyWavelets change

**Comment by terencehonles** [comment]
> Azure passed just by restarting the job. It looks the change here now need to be rebased, though

It looks like some of the job runners for Azure Pipelines haven't been refreshed for the new build tools. I'm not sure if there's a flag you can specify to pick the specific version.

It looks like you would need to specify https://github.com/actions/virtual-environments/blob/14b4cec214c5a521821055add04699f48e72bf75/images/linux/Ubuntu1804-README.md which is image version 20201012.1 or greater

**Review by emmanuelle — APPROVED** [review]
Thanks @terencehonles looks good to me! We cannot test the Travis changes any more but your changes will be useful to port as we are migrating to Github Actions.

File: .appveyor.yml [hunk 1]
```diff
@@ -29,23 +35,23 @@ install:
...
+  - pip install %PIP_FLAGS% -r requirements/default.txt
+  - pip install %PIP_FLAGS% -r requirements/build.txt
   - python setup.py bdist_wheel bdist_wininst
   - ps: "ls dist"
 
   # Install the generated wheel package to test it.
   - pip install %PIP_FLAGS% --no-index --find-links dist/ scikit-image
   # Install the test dependencies
-  - pip install %PIP_FLAGS% --retries 3 -r requirements/test.txt
+  - pip install %PIP_FLAGS% -r requirements/test.txt
```
Comment (jni) [inline .appveyor.yml:54]: Why are these removed?

Comment (terencehonles) [reply .appveyor.yml:54]: The [default now is 5](https://pip.pypa.io/en/stable/reference/pip/#cmdoption-retries) and I don't think the author wanted *less* retries

**Comment by hmaarrfk** [comment]
Cool!
